django/db/models/query.py

```diff
@@ -544,19 +545,31 @@ async def aiterator(self, chunk_size=2000):
         An asynchronous iterator over the results from applying this QuerySet
         to the database.
         """
-        if self._prefetch_related_lookups:
-            raise NotSupportedError(
-                "Using QuerySet.aiterator() after prefetch_related() is not supported."
-            )
         if chunk_size <= 0:
             raise ValueError("Chunk size must be strictly positive.")
         use_chunked_fetch = not connections[self.db].settings_dict.get(
             "DISABLE_SERVER_SIDE_CURSORS"
         )
+
+        chunk = deque()
+        num_items = 0
         async for item in self._iterable_class(
             self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size
         ):
-            yield item
+            chunk.append(item)
+            num_items += 1
+            if num_items == chunk_size:
+                await sync_to_async(prefetch_related_objects)(
+                    chunk, *self._prefetch_related_lookups
+                )
+                for each_item in chunk:
+                    yield each_item
+                chunk.clear()
+                num_items = 0
+
+        if num_items:
```

Comment:
You'll want to prefetch related as well here if there are some times left in the buffer

```suggestion
        if chunk:
           await sync_to_async(prefetch_related_objects)(
                chunk, *self._prefetch_related_lookups
            )
```